### PR TITLE
Take cleanup

### PR DESCRIPTION
**Pull request by ith-harvey**
* Adjusted when `liquidationBondCleanup` is updated
* Reassigned `collateralTaken_` after possible `RepayAmount_` conditional

### CODE REVIEW

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol
Comment (EdNoepel) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol:531]: An auction was kicked, and 2 collateral tokens are under liquidation.  Lender runs a full `take` for all the collateral, but only gets 0.127 of it?  Why?

Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol:531]: I asked Matt the same question. This is to protect the borrower... The amount of collateral that we allow the taker to take is restricted by the amount of debt that needs to be covered at the current `auctionPrice` . In this case the borrower only had sub `20.0` of debt, therefore we only allow the taker to take `0.127` collateral because the `aucitonPrice` is so high.

File: src/libraries/Auctions.sol
Comment (ith-harvey) [inline src/libraries/Auctions.sol:288]: Instead of calculating `collateralTaken_` once on LN 210 we calculate it here and on LN 201 to increase the accuracy of the `collateralTaken_` value. (division twice is BAD)

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol
Comment (EdNoepel) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol:531]: Indeed.  I bumped it to 4 hours and got a quarter of the collateral requested.
How can we query the neutral price of the liquidation?

Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol:531]: Hmm correct me if I'm wrong but the same UX would happen if the user were to pass in `maxQuotetoken`... so I think it's safe to say that `maxCollateral` isn't an issue.

Well since NP is calculated by -> `int256 neutralPrice = int256(Maths.wmul(mompFactor_, inflatorSnapshot_));`
, we could have the UI / SDK get the mompFactor of the borrower in liquidation and multiply that by the PoolState.inflator (both of these values would be out of date so to speak as the pool level inflator is only congruent with the last time the pool was touched... ). That could at least provide the taker with a worst expected take (worst expected swap of QT for CT base d on old `neutralPrice` value). Admittedly I don't like that UX....

Comment (EdNoepel) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol:531]: Agreed; I was confusing the MOMP (attached to the liquidation) with the MOMP factor (attached to the borrower).  If I understand correctly, the former is the neutral price and requires no inflator adjustment.

File: src/libraries/Auctions.sol
Comment (grandizzy) [inline src/libraries/Auctions.sol]: This can be removed

File: src/libraries/Auctions.sol
Comment (grandizzy) [inline src/libraries/Auctions.sol]: we have to assert / cover `liquidationBondEscrowed` accumulator changes in our forge tests (maybe as part of `_assertAuction`)

**Review by grandizzy — CHANGES_REQUESTED** [review]

File: src/libraries/Auctions.sol
Comment (EdNoepel) [reply src/libraries/Auctions.sol]: If we make changes to `_assertAuction`, I recommend we assert on the auction price as well.

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol:531]: No the latter, as per the V10 docs: `The actual neutral price for the loan is this factor multiplied by the current borrowerInflator.`

File: src/libraries/Auctions.sol
Comment (ith-harvey) [reply src/libraries/Auctions.sol]: resolved -> https://github.com/ajna-finance/contracts/pull/296/commits/8cc44ade159bcc35c8f3ba68dfadd7d0e0c3aa4a

File: src/libraries/Auctions.sol
Comment (ith-harvey) [reply src/libraries/Auctions.sol]: resolved -> https://github.com/ajna-finance/contracts/pull/296/commits/8cc44ade159bcc35c8f3ba68dfadd7d0e0c3aa4a

File: src/base/Pool.sol
Comment (EdNoepel) [inline src/base/Pool.sol]: `auctionInfo` should only return data which pertains to an auction identified by the specified borrower.  It should not return the pool-level `totalBondEscrowed`.  `DSTestPlus._assertAuction` should query that separately.

**Review by EdNoepel — CHANGES_REQUESTED** [review]

File: src/base/Pool.sol
Comment (ith-harvey) [reply src/base/Pool.sol]: resolved -> https://github.com/ajna-finance/contracts/pull/296/commits/bb4faa208ab8e3c7e4b8e677af5f7a72ae06554d

Comment (EdNoepel) [reply src/base/Pool.sol]: 👌

**Review by EdNoepel — APPROVED** [review]
Looks good.  👍

**Comment by ith-harvey** [comment]
I'll wait to merge until after @grandizzy PR

**Comment by grandizzy** [comment]
> I'll wait to merge until after @grandizzy PR

It is merged now, there are merging conflicts in this PR though :(

**Review by grandizzy — APPROVED** [review]
good to go after resolving merge PR

**Review by EdNoepel — APPROVED** [review]
Still looks good.

**Pull request merged**
